packages/ui_v2/src/components/commonV2/dao/GenesisProtocolEditor.tsx

```diff
@@ -11,18 +12,32 @@ export default function GenesisProtocolEditor(props: Props) {
   const { form, editable } = props;
   const formState = form.$ as any;
 
-  // TODO: make this the default behaviour of all form components
-  // All editors should just be this, and the fields themselves should contain
-  // a type. Then overrides can be made.
-  return (
-    <>
-      {Object.keys(formState).map((propName: string, index: number) => (
+  const votingMachingParamsList: Array<any> = [];
+  let paramsNames = Object.keys(formState);
+  let i: any;
+  for (i in paramsNames) {
```

Comment:
This code must be refactored, right now I just made it useful but it will be improved

Reply:
Done